src/routes/safe/store/actions/transactions/fetchTransactions/loadAllTransactions.ts

```diff
@@ -0,0 +1,81 @@
+import { List, Map } from 'immutable'
+import axios, { AxiosResponse } from 'axios'
+
+import { getAllTxServiceUriTo, getTxServiceHost } from 'src/config'
+import { checksumAddress } from 'src/utils/checksumAddress'
+import { Transaction } from '../../../models/types/transactions'
+
+type ServiceUriParams = {
+  safeAddress: string
+  limit?: number
+  offset?: number
+  orderBy?: string // todo: maybe this should be key of MultiSigTransaction | keyof EthereumTransaction
+  queued?: boolean
+  trusted?: boolean
+}
+
+type EndpointResponse = {
+  count: number
+  next?: string
+  previous?: string
+  results: Transaction[]
+}
+
+const getAllTransactionsUrl = (safeAddress: string) => {
```

Comment:
shouldn't it receive `(params: ServiceUriParams)` instead of only safeAddress? 
Also, I think `limit` and `offset` should be required.

Reply:
yep, I updated the params but not for `getAllTransactionsUrl`, for `fetchAllTransactions` and `loadAllTransactions`

Reply:
Which limit and offset default values do you think it's ok?

Reply:
we can ask @lukasschor but I would suggest bringing 20 Txs per round sounds reasonable.

Reply:
I'm curious how are you going to implement pagination?

Reply:
Given that cancellation transactions are not shown as a separate row

Reply:
As far as we discussed internally we will have a new txs endpoint that will give us a list of objects, each object should represent a row (with the cancellations and the transactions related to that row). So basically the pagination will be independent of the data, each value of the array should be a row

Example: 

- Tx1 has one cancellation transaction 
- Tx2 is confirmed without cancellation

The endpoint now give us an array like: `[tx1, cancellation, tx2]` ->, in this case, the pagination should take care of the data inside the array

With the new endpoint we will receive something like: [{Tx1, cancellation}, {tx2}] -> now the pagination is direct

Correct me if I made a mistake @nicosampler  @fernandomg

Reply:
@Agupane are you talking about the new endpoint on transaction service or the one on a middle layer backend we're planning to make?

Reply:
the new middleware

Reply:
I'm not sure if we should count on it now.. It's not even planned or estimated yet

Reply:
As I understood yesterday, we don't have to make this work with FE logic, so we are assuming that at some moment we will have this functionality. 
I think having normal pagination (without any trick in the middle) is the safest way to advance.

Reply:
my two cents:

I'd aim for small implementations/updates/upgrades based on the decisions that we've already made. In this case, I'd just try to add as little code as possible so removing it is easy and as safe as human possibly.

- If later on we decide not to implement this 'intermediate layer', then we code pagination in the FE
- If later on we decide to implement this 'intermediate layer', then we are just fine.

takeaway: shall we leave pagination out of the equation for now? If not, why don't we implement as little as possible on this matter?

@lukasschor CC'ing you in case you wanna jump in